### PR TITLE
add an example of using `sizeof` on a struct with padding

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

File: base/essentials.jl [hunk 1]
```diff
@@ -539,6 +539,17 @@ julia> sizeof(1.0)
...
 julia> sizeof(collect(1.0:10.0))
 80
+
+julia> struct StructWithPadding
+           x::Int64
+           flag::Bool
+       end
+
+julia> sizeof(StructWithPadding) # not the sum of `sizeof` of fields due to padding
+16
```
Comment (vtjnash) [inline base/essentials.jl:549]: Keno had proposed changing this (https://github.com/JuliaLang/julia/pull/46322), though we didn't finish

Comment (vtjnash) [reply base/essentials.jl:549]: The exact number here is definitely an implementation detail. For example, on 32-bit platforms this returns:
```
julia> sizeof(StructWithPadding)
12
```

**Comment by PallHaraldsson** [comment]
Also mention the order matters for size:
```
julia> struct StructWithPadding5
                  x::Int16
                  flag::UInt8
                  flag2::Bool
               end

julia> struct StructWithPadding6
         flag::UInt8
         x::Int16
         flag2::Bool
       end
```

**Comment by KristofferC** [comment]
I think this is an improvement since it makes you aware the `sizeof` is not necessarily the sum of field sizes.

**Pull request merged**